src/MacroTools/Extensions/RectangleExtensions.cs

```diff
@@ -66,6 +66,7 @@ public static void CleanupNeutralPassiveUnits(this Rectangle area, NeutralPassiv
         }
 
         if (cleanupType == NeutralPassiveCleanupType.RemoveUnits || unit.IsType(UNIT_TYPE_STRUCTURE))
+          if (unit.GetTypeId() != FourCC("ngme"))
```

Comment:
As discussed on Discord, let's make this logic generic rather than relying on a hardcoded FourCC. The other one I put in this function is not great either. 

Instead we can check to see if the unit being iterated over is `NeverDelete` and invulnerable. If that's the case, we don't need to do anything to clean the unit up. We'll also have to make sure Goblin Merchant is registered as `NeverDelete` in `UnitTypeConfig`.